include/ideep/operators/matmul.hpp

```diff
@@ -513,9 +515,20 @@ enum task_type {
      std::vector<int64_t> dst_strides = (ndims == 3) ?
          std::vector<int64_t>({dst_dims[2]* dst_dims[1], dst_dims[1], 1}) :
          std::vector<int64_t>({dst_dims[1], 1});
-     tensor::desc dst_desc = is_dynamic ?
-         tensor::desc(dst_dims, dst_data_type, dst_strides) :
-         tensor::desc(dst_dims, dst_data_type, tag::any);
+
+     tensor::desc dst_desc;
+     if (cpuinfo_initialize() && cpuinfo_has_arm_bf16()) {
```

Comment:
I think we don't need to depend on the cpuinfo library, OneDND has a API check the ISA: https://github.com/oneapi-src/oneDNN/blob/f03046f2331a586905b2d41ede028af52ffe9590/include/oneapi/dnnl/dnnl.hpp#L13075, and the ARM ISA list is https://github.com/oneapi-src/oneDNN/blob/master/src/cpu/aarch64/cpu_isa_traits.hpp#L47, which has a mapping to X86 ISA: https://github.com/oneapi-src/oneDNN/blob/master/include/oneapi/dnnl/dnnl_types.h#L3174, you can see the x86 bf16 check: https://github.com/intel/ideep/blob/pytorch-internal/include/ideep/abstract_types.hpp#L70.

Reply:
Hi @XiaobingSuper , the fix is not specific to bf16 kernels. If the tensor dimensions based dest descriptor definition is acceptable for x86, we can make this change without any conditional checks. Could you please review and comment?

Reply:
@Xia-Weiwen , could you help check whether we need set dst_desc to format_any or not?  if not, the platform check will not be needed.

Reply:
@Xia-Weiwen , if we don't explicitly set the format to ::any, the ideep tensor class populates it with the right format based on the dimensions.

Reply:
I think it's OK to remove `tag::any` if all PyTorch/IPEX UTs pass.

Reply:
thanks! does any CI cover these tests or is it all manual?

Reply:
I will update the PR to make the change platform agnostic.

Reply:
> thanks! does any CI cover these tests or is it all manual?

There is no CI for ideep only. We need to do it manually with IPEX and PyTorch

Reply:
Please let me know the next steps here. I'm happy to help with any test coverage to accelerate this pr review and merging.

Reply:
> Please let me know the next steps here. I'm happy to help with any test coverage to accelerate this pr review and merging.

Hi @snadampal. Are you going to use it in stock PyTorch or internal PyTorch? For stock PyTorch, we probably won't update ideep inside it before 1.13 release. For internal PyTorch, changes should be made to branch ideep_dev.
In the future (probably after PyTorch 1.13 release), we are going to use branch ideep_pytorch for stock PyTorch.

Reply:
For manual tests, can run UTs of PyTorch and IPEX.
PyTorcch's UTs are in `<pytorch source root>/test/` by `$python xxx.py`
- test_ops.py
- test_mkldnn*.py
- test_quantization.py

For IPEX, go to `<IPEX source root>/tests/cpu/` then run `$pytest test_*`

Reply:
thanks @Xia-Weiwen , I have a PR on `ideep_dev` branch as well which I think can be targeted for PT1.13. I will update the test results in both the PRs.

Reply:
I see. Also @Guobing-Chen @XiaobingSuper @yanbing-j here. Can we make it to land in PT 1.13? If so, should we update pytorch_dnnl or ideep_pytorch?

Reply:
@Xia-Weiwen , I have executed the PyTorch UT on AWS c6i (x86) instances and here are the results. In summary there is no regression with this PR, in fact 3 tests are additionally passing on test_ops.py and 1 additional test passing from test_quantization compared to the PyTorch master. I will keep you posted on the IPEX UT.

**PyTorch master:**
test_ops.py:

> Ran 25514 tests in 1062.573s
>
> FAILED (failures=20, errors=69, skipped=8001, expected failures=276)

test_mkldnn.py:

> Ran 67 tests in 19.717s
>
> OK

test_mkldnn_fusion.py:

> Ran 3 tests in 77.529s
> 
> OK

test_mkldnn_verbose.py:

> Ran 2 tests in 1.657s
> 
> OK

test_quantization.py:

> Ran 930 tests in 382.495s
>
> FAILED (failures=1, skipped=65)

**PyTorch master + with this pr:**
test_ops.py:

> Ran 25514 tests in 1063.838s
> 
> FAILED (failures=17, errors=69, skipped=8001, expected failures=276)

test_mkldnn.py:

> Ran 67 tests in 19.446s
> 
> OK

test_mkldnn_fusion.py:

> Ran 3 tests in 77.601s
> 
> OK


test_mkldnn_verbose.py:

> Ran 2 tests in 1.659s
> 
> OK

test_quantization.py:

> Ran 930 tests in 383.025s
> 
> OK (skipped=65)